### PR TITLE
Add interfaces to Item and Property

### PR DESCRIPTION
**Pull request by Benestar**
With this patch applied, Item and Property implement the term related
interfaces LabelsProvider, DescriptionsProvider and AliasesProvider.

Note that the signature of getLabels and getDescriptions changed and those
methods now return TermList objects instead of plain arrays.

Depends on #631 

**Note that this shouldn't be merged before we are sure the next release will be 6.0.0**

Bug: [T126152](https://phabricator.wikimedia.org/T126152)

### CODE REVIEW

**Comment by JeroenDeDauw** [comment]
Had a quick look. +1

File: tests/unit/Entity/PropertyTest.php [hunk 1]
```diff
@@ -229,4 +320,18 @@ public function testPropertyWithStatementsIsNotEmpty() {
 		$this->assertFalse( $property->isEmpty() );
 	}
 
+	public function testCopy() {
```
Comment (thiemowmde) [inline tests/unit/Entity/PropertyTest.php]: Unrelated. Please move to a separate patch.

File: tests/unit/Entity/PropertyTest.php [hunk 2]
```diff
@@ -7,32 +7,20 @@
...
+use Wikibase\DataModel\Term\Term;
+use Wikibase\DataModel\Term\TermList;
 
 /**
  * @covers Wikibase\DataModel\Entity\Property
- * @covers Wikibase\DataModel\Entity\Entity
- *
- * @group Wikibase
- * @group WikibaseProperty
- * @group WikibaseDataModel
```
Comment (thiemowmde) [inline tests/unit/Entity/PropertyTest.php]: What's wrong with these tags? At least the `@group WikibaseDataModel` should stay.

File: tests/unit/Entity/ItemTest.php [hunk 3]
```diff
@@ -368,4 +460,18 @@ public function testNotEquals( Item $firstItem, Item $secondItem ) {
 		$this->assertFalse( $secondItem->equals( $firstItem ) );
 	}
 
+	public function testCopy() {
```
Comment (thiemowmde) [inline tests/unit/Entity/ItemTest.php]: Unrelated. Please move to a separate patch.

File: tests/unit/Entity/ItemTest.php [hunk 4]
```diff
@@ -4,43 +4,26 @@
...
- * @covers Wikibase\DataModel\Entity\Entity
- *
- * Some tests for this class are located in ItemMultilangTextsTest,
- * ItemNewEmptyTest and ItemNewFromArrayTest.
- *
- * @since 0.1
- *
- * @group Wikibase
- * @group WikibaseItem
- * @group WikibaseDataModel
```
Comment (thiemowmde) [inline tests/unit/Entity/ItemTest.php]: As below.

File: src/Entity/Property.php [hunk 5]
```diff
@@ -251,4 +292,15 @@ public function setStatements( StatementList $statements ) {
...
 	}
 
+	/**
+	 * @see EntityDocument::copy
+	 *
+	 * @since 0.1
+	 *
+	 * @return self
+	 */
+	public function copy() {
```
Comment (thiemowmde) [inline src/Entity/Property.php]: Unrelated. Also conflicts with #626. Please move to a separate patch.

File: src/Entity/Item.php [hunk 6]
```diff
@@ -319,4 +360,15 @@ public function equals( $target ) {
...
 	}
 
+	/**
+	 * @see EntityDocument::copy
+	 *
+	 * @since 0.1
+	 *
+	 * @return self
+	 */
+	public function copy() {
```
Comment (thiemowmde) [inline src/Entity/Item.php]: Unrelated. Also conflicts with #626. Please move to a separate patch.

File: RELEASE-NOTES.md [hunk 7]
```diff
@@ -1,5 +1,17 @@
 # Wikibase DataModel release notes
 
+## Version 6.0.0 (dev)
+
+* Removed `Entity` class (deprecated since 1.0)
```
Comment (thiemowmde) [inline RELEASE-NOTES.md]: I don't understand why we need to kill the class and refactor everything in a single step, in a patch that exclusively talks about new interfaces. We can easily introduce the interfaces and only remove the conflicting methods from Entity.

Comment (Benestar) [inline RELEASE-NOTES.md]: You havev to read the patches in the correct order... This pull request includes three commits and it states in the description on which pull request it depends. Unfortunately, github doesn't handle patch chains as nice as gerrit.

File: tests/unit/Entity/PropertyTest.php (hunk 2 above)
Comment (Benestar) [inline tests/unit/Entity/PropertyTest.php]: We don't use `@group` annotations in most of our tests. Either use them everywhere or remove them, everything else is just confusing.

Comment (thiemowmde) [inline tests/unit/Entity/PropertyTest.php]: This is unrelated to all tree commits in this chain. Please revert this and upload a separate patch. Personally, I prefer to have the said `@group WikibaseDataModel` in all tests.

**Comment by thiemowmde** [comment]
Makes soooooooooo much sense now. @Benestar, thanks a lot for making these commits so much fun to review!

**Comment by Benestar** [comment]
> Does this mean Fingerprint can also implement LabelsProvider, DescriptionsProvider and AliasesProvider now?

It already does :)
